src/controllers/admin.controller.js

```diff
@@ -45,32 +49,36 @@ adminCtrl.addCourse = async (req, res) => {
         });
     } catch (err) {
         console.log(err.response);
-        if(err.response && err.response.data){
+        if (err.response && err.response.data) {
             let errorMsg = err.response.data.message;
             req.flash("error_msg", errorMsg);
         };
     };
-  res.redirect("/admin/course");
+    res.redirect("/admin/course");
 };
 
 // AT-UNIVERSITY - Admin - Render Edit Course Form
 adminCtrl.renderEditCourseForm = async (req, res) => {
-    console.log("--> adminCtrl.renderEditCourseForm");
-    let courseId = req.params.id;
-    universityServiceAPI.getCourseById(courseId);
-
-    // Temporary code to retrive information about the course
-
-    let responseCourses = await universityServiceAPI.getAllCourses();
-
-    courseDetails = responseCourses.data.filter(function (c) { return c._id == courseId; });
-
-    console.log("One course found", courseDetails[0]);
+    try {
+        console.log("---> adminCtrl.renderEditCourseForm");
+        let courseId = req.params.id;
+        universityServiceAPI.getCourseById(courseId);
 
-    // ---
+        // Temporary code to retrive information about the course
 
-    res.render("admin/course/edit-course", courseDetails[0]);
-};
+        let responseCourses = await universityServiceAPI.getAllCourses();
+        courseDetails = responseCourses.data.filter(function (c) { return c._id == courseId; });
+        console.log("One course found", courseDetails[0]);
+        res.render("admin/course/edit-course", courseDetails[0]);
+    } catch (err) {
+        console.log(err.response);
+        if (err.response && err.response.data) {
+            let errorMsg = err.response.data.error;
+            req.flash("error_msg", errorMsg, " Try again later");
+            res.redirect("/admin");

```

Comment:
is that correct?

Reply:
Yes 😄

Reply:
I think no, should be `res.redirect("/admin/course");`

Reply:
ready 😄